File: src/main/java/ma/fstg/gestionprojets/services/FactureService.java
package ma.fstg.gestionprojets.services;
import ma.fstg.gestionprojets.dto.request.FactureRequest;
import ma.fstg.gestionprojets.dto.response.FactureResponse;
import java.util.List;
public interface FactureService {
    FactureResponse create(Long phaseId, FactureRequest req);
    FactureResponse update(Long id, FactureRequest req);
    FactureResponse getById(Long id);
    List<FactureResponse> getAll();
    void delete(Long id);
}

File: gestion-projets-frontend/src/services/factureService.ts
import api from '../utils/axiosConfig';

export const factureService = {
  getAll: () => api.get('/factures'),
  getById: (id: number) => api.get(`/factures/${id}`),
  create: (phaseId: number, data: any) => api.post(`/phases/${phaseId}/facture`, data),
  update: (id: number, data: any) => api.put(`/factures/${id}`, data),
  delete: (id: number) => api.delete(`/factures/${id}`),
};
